Position position
float64 v
EulerAngle euler_angle